packages/adapter-ccxt/src/validation/AdapterConfigValidator.ts

```diff
@@ -2,70 +2,85 @@ import { injectable } from "tsyringe";
+   }
 
-         for (const exchange of validationResult.right.config
-            .exchanges as Array<{
-            id: string;
-            api_key: string;
-            api_secret: string;
-         }>) {
-            const exchangeId = exchange.id.trim().toLowerCase();
+   private extractExchangeCredentials(
+      exchanges: Array<{ id: string; api_key: string; api_secret: string }>
+   ): Map<ExchangeId, ApiCredentials> {
+      const exchangeCredentials: Map<ExchangeId, ApiCredentials> = new Map();
 
-            // Make sure the exchange is supported
-            if (!Object.values(ExchangeId).includes(exchangeId as ExchangeId)) {
-               throw new ValidationError(
-                  `${INVALID_ADAPTER_CONFIGURATION}: ${UNSUPPORTED_EXCHANGE_ERROR(
-                     exchangeId
-                  )}`
-               );
-            }
+      for (const exchange of exchanges) {
+         this.validateExchangeId(exchange.id, exchangeCredentials);
 
-            // Make sure the exchange ID is unique, e.g. no duplicate exchange IDs in the config
-            if (exchangeCredentials.has(exchangeId as ExchangeId)) {
-               throw new ValidationError(
-                  `${INVALID_ADAPTER_CONFIGURATION}: ${DUPLICATE_EXCHANGE_ID_ERROR(
-                     exchange.id
-                  )}`
-               );
-            }
+         const exchangeApiCredentials = this.extractApiCredentials(exchange);
+         exchangeCredentials.set(
+            exchange.id.trim().toLowerCase() as ExchangeId,
+            exchangeApiCredentials
+         );
+      }
+      return exchangeCredentials;
+   }
 
-            const exchangeApiCredentials: ApiCredentials = {
-               apiKey: exchange.api_key.trim(),
-               apiSecret: exchange.api_secret.trim(),
-            };
+   private validateExchangeId(
+      id: string,
+      exchangeCredentials: Map<ExchangeId, ApiCredentials>
+   ): void {
+      const exchangeId = id.trim().toLowerCase();
 
-            exchangeCredentials.set(
-               exchangeId as ExchangeId,
-               exchangeApiCredentials
-            );
-         }
+      if (!Object.values(ExchangeId).includes(exchangeId as ExchangeId)) {
+         throw new ValidationError(
+            `${ERR_INVALID_ADAPTER_CONFIG}: ${ERR_UNSUPPORTED_EXCHANGE(
+               exchangeId
+            )}`
+         );
+      }
 
-         return new CCXTAdapterConfig(exchangeCredentials);
-      } else if (isLeft(validationResult)) {
+      if (exchangeCredentials.has(exchangeId as ExchangeId)) {
          throw new ValidationError(
-            INVALID_ADAPTER_CONFIGURATION,
-            PathReporter.report(validationResult)
+            `${ERR_INVALID_ADAPTER_CONFIG}: ${ERR_DUPLICATE_EXCHANGE_ID(id)}`
          );
-      } else {
-         throw new Error(INVALID_ADAPTER_CONFIGURATION);
       }
    }
+
+   private extractApiCredentials(exchange: {
+      api_key: string;
+      api_secret: string;
```

Comment:
> I think we should always use camel-case for these (even in the yaml) and then the io-ts codecs should be exposed as types as well with the `t.TypeOf` helper, because I saw in some places that we have both the codec with `_` in params and then a separate manually created type that gets populated and I think that can get messy and confusing, so having just codecs + auto-generated types from the codecs that get used in the application would be best imo.

In agreement with the camel case. There is only one manually created type, the `CCXTAdapterConfig`, this was left over from the initial code before I found the TypeOf helper. Will update it too 👍.